src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/KerberosTicketRetrievalCallback.cs

```diff
@@ -0,0 +1,13 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+
+namespace Microsoft.Data.SqlClient
+{
+        /// <summary>
+        /// A callback to provide kerberos ticket for given service principal name (SPN) on demand.
+        /// </summary>
+        /// <returns></returns>
+        public delegate byte[] KerberosTicketRetrievalCallback(string serverPrincipalName);
```

Comment:
Does this need to be a new type, or is there a reason to use a type over a `Func<byte[],string>` ? 

I'm also no sure about the naming. Is it really a callback? it feels more like a provider or an adapter. The user will set the this property of the connection to and it will be called by the connection to get the ticket to use. So we're not really calling back to anything instead we're providing a way to get an external value by some means. `KerberosTicketProvider` ? something like that?

Reply:
This follows conventions for this library ( https://github.com/dotnet/SqlClient/search?q=callback ). Team is also exploring changing this into adapter pattern design already used in FedAuth flow.